package_control/distinfo.py

```diff
@@ -0,0 +1,381 @@
+import base64
+import hashlib
+import os
+import re
+import sys
+
+from . import __version__ as pc_version
+
+_dist_info_pattern = re.compile(
+    r'''(?x)
+    (?P<name>.+?)-(?P<version>
+        (?P<major>[0-9]+)
+      \.(?P<minor>[0-9]+)
+      \.(?P<patch>[0-9]+)
+      (?:\-(?P<prerelease>(?:[0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*)?))?
+      (?:\+(?P<build>(?:[0-9A-Za-z-]+(?:\.[0-9A-Za-z-]+)*)?))?
+    )'''
+)
+
+
+def match_dist_info_dir(dir_name, library_name):
+    match = _dist_info_pattern.match(dir_name)
+    if match and match['name'] == library_name:
+        return match
+    return False
+
+
+def find_dist_info_dir(install_root, library_name):
+    """
+    Find the .dist-info directory for a given library.
+
+    :param install_root:
+        A unicode string of the dir the package is installed in
+
+    :param library_name:
+        A unicode string of the library name
+
+    :returns:
+        A unicode string of the .dist-info directory.
+
+    :raises:
+        FileNotFoundError if no .dist-info directory was found.
+    """
+
+    for dir_name in os.listdir(install_root):
+        if match_dist_info_dir(dir_name, library_name):
+            return DistInfoDir(install_root, dir_name)
+    raise FileNotFoundError('Library {} not installed!'.format(library_name))
+
+
+def list_dist_info_dirs(install_root):
+    for dir_name in os.listdir(install_root):
+        if dir_name.endswith('.dist-info'):
+            yield DistInfoDir(install_root, dir_name)
+    return False
+
+
+class DistInfoDir:
+
+    def __init__(self, install_root, dist_info_dir):
+        self.install_root = install_root
```

Comment:
It would probably good to add a docstring to explain what these parameters should be